### PR TITLE
Add simple numerical data exercise.

### PR DESCRIPTION
**Pull request by lesteve**
This is my attempt at a simple numerical data with k-neighbors classifier.

Note: I originally wanted to do the `n_neighbors=1` 100% accuracy on the train set but not on the test set. Unfortunately life is always more complicated than you think 😢. Because there are many duplicated rows (in this numerical-only dataset, we have 4 features `age`, `hours-per-week`, `capital-gain`, `capital-loss`) with different classes the accuracy is not 100% on the train set.

Not only it does not overfit with `n_neighbors=1` but it actually even underfits (it performs about the same as a majority class predictor both on the train and test set)  for some reason that I let as an exercise for the astute reader.

### CODE REVIEW

File: python_scripts/02_numerical_pipeline_sol_00.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+first_target_values = target.iloc[:10]
+first_target_values
+
+# %%
+number_of_correct_predictions = (
+    first_predictions == first_target_values).sum()
+number_of_predictions = len(first_predictions)
+print(
+    f"{number_of_correct_predictions}/{number_of_predictions} "
+    "of predictions are correct")
```
Comment (glemaitre) [inline python_scripts/02_numerical_pipeline_sol_00.py:80]:
```suggestion
print(f"{number_of_correct_predictions}/"f
      f"{number_of_predictions} of predictions are correct")
```

File: python_scripts/02_numerical_pipeline_sol_00.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+# data. Do they match the actual target values?
+
+# %%
+first_data_values = data.iloc[:10]
+first_predictions = model.predict(first_data_values)
+first_predictions
+
+# %%
+first_target_values = target.iloc[:10]
+first_target_values
```
Comment (glemaitre) [inline python_scripts/02_numerical_pipeline_sol_00.py:72]: To avoid the issue of the rendering (Series vs. array), we can use a `for` loop a bit similar to:

```python
for sample_idx in range(5):
    print(f"Sample #{sample_idx} - true: {target[samples_idx]} vs. "
          f"predicted: {predicted_target[sample_idx}")
```

**Review by glemaitre — APPROVED** [review]
LGTM

File: python_scripts/02_numerical_pipeline_sol_00.py (hunk 1 above)
Comment (lesteve) [reply python_scripts/02_numerical_pipeline_sol_00.py:80]: Interesting I slightly prefer my way I don't really know why ...

File: python_scripts/02_numerical_pipeline_sol_00.py (hunk 2 above)
Comment (lesteve) [reply python_scripts/02_numerical_pipeline_sol_00.py:72]: I think good enough for now, you are right that this is not such a big deal ...

**Comment by lesteve** [comment]
OK let's merge this one, thanks for having a look!

File: python_scripts/02_numerical_pipeline_sol_00.py (hunk 1 above)
Comment (glemaitre) [reply python_scripts/02_numerical_pipeline_sol_00.py:80]: Maybe because it is closer to black :)
Throughout the notebooks, the style is to use something the most compact. But this is nitpicking.

**Comment by lfarhi** [comment]
Can you generate the associated notebook 02_numerical_pipeline_ex_00.ipynb ?

**Comment by lesteve** [comment]
Good point, I'll do this.

**Comment by lesteve** [comment]
Done in 5ca67435cfba71b7527437b996e40382872c8892

**Comment by lfarhi** [comment]
Thanks
